File: im-backend/src/main/java/com/im/backend/entity/ConversationFolder.java
package com.im.backend.entity;

import lombok.*;
import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "conversation_folders")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ConversationFolder {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "user_id", nullable = false)
    private Long userId;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "icon")
    private String icon;

    @Column(name = "color")
    private String color;

    @Column(name = "sort_order")
    private Integer sortOrder;

    @Column(name = "is_collapsed")
    private Boolean isCollapsed;

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        if (sortOrder == null) sortOrder = 0;
        if (isCollapsed == null) isCollapsed = false;
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: im-mobile/lib/models/conversation_folder.dart
class ConversationFolder {
  final int id;
  final int userId;
  final String name;
  final String? icon;
  final String? color;
  final int sortOrder;
  final bool isCollapsed;
  final DateTime createdAt;
  final DateTime? updatedAt;

  ConversationFolder({
    required this.id,
    required this.userId,
    required this.name,
    this.icon,
    this.color,
    required this.sortOrder,
    required this.isCollapsed,
    required this.createdAt,
    this.updatedAt,
  });

  factory ConversationFolder.fromJson(Map<String, dynamic> json) {
    return ConversationFolder(
      id: json['id'],
      userId: json['userId'],
      name: json['name'],
      icon: json['icon'],
      color: json['color'],
      sortOrder: json['sortOrder'] ?? 0,
      isCollapsed: json['isCollapsed'] ?? false,
      createdAt: DateTime.parse(json['createdAt']),
      updatedAt: json['updatedAt'] != null ? DateTime.parse(json['updatedAt']) : null,
    );
  }
}
